lib/cryptozoologist/generator.rb

```diff
@@ -48,12 +48,71 @@ def random
 
     # Generates a string for a street address with a number and street. Only
     # uses animal dictionaries and does not respect config exclusions.
-    # 
+    #
     def street_address
       number = rand(1..9000)
       street = Dictionary.animals.sample
       street = street.split(" ").map! {|word| word.capitalize! }.join(" ")
       "#{number} #{street} #{Dictionary.addresses.sample}"
     end
+
+    # Generates a string for a U.S. state name which is partially replaced by
+    # alliterative words from other dictionaries (e.g. "Oregon" can become "Oregoose" or "Orabbit")
+    # Does not respect config exclusions
+    def state
+      libraries = [:animals, :clothing, :colors]
+      state_name = Dictionary.states.sample
+      has_two_words = state_name.index(" ")
+
+      if has_two_words
+        final_word = handle_two_word(state_name, libraries)
+      else
+        final_word = handle_one_word(state_name, libraries)
+      end
+
+      final_word
+    end
+
+    private
+
+    #completely replaces one random word from the state with an alliterative word from libraries
+    def handle_two_word(state_name, libraries)
+      base_state_words = state_name.split(" ")
+      replace_index = rand(0..1)
+      base_state_words[replace_index] = get_alliteration(libraries, base_state_words[replace_index][0]).capitalize
+      return base_state_words.join(" ")
+    end
+
+    #selects a random key_letter from state_name, and finds an alliterative word from libraries for replacement
+    def handle_one_word(state_name, libraries)
+      key_letter_index = rand(0...state_name.length-1)
+      insert_word = get_alliteration(libraries, state_name[key_letter_index]).downcase
+
+      #if the randomly selected word from libraries has more letters than the substring from key_letter_index to state_name.length
+      #completely replace that substring
+      if insert_word.length > state_name.length - key_letter_index && key_letter_index > 0
```

Comment:
can you add some parenthesis around this? it's difficult for me to discern the order of operations here with all the comparisons and mathing